modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -724,5 +718,306 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
+   */
+  private getHashStringAndDerivationPath(
+    txRequest: TxRequest,
+    requestType: RequestType = RequestType.tx
+  ): { hashBuffer: Buffer; derivationPath: string } {
+    let txToSign: string;
+    let derivationPath: string;
+    if (requestType === RequestType.tx) {
+      assert(txRequest.transactions && txRequest.transactions.length === 1, 'Unable to find transactions in txRequest');
+      txToSign = txRequest.transactions[0].unsignedTx.signableHex;
+      derivationPath = txRequest.transactions[0].unsignedTx.derivationPath;
+    } else if (requestType === RequestType.message) {
+      throw new Error('MPCv2 message signing not supported yet.');
+    } else {
+      throw new Error('Invalid request type, got: ' + requestType);
+    }
+
+    let hash: Hash;
+    try {
+      hash = this.baseCoin.getHashFunction();
+    } catch (err) {
+      hash = createKeccakHash('keccak256') as Hash;
+    }
+    const hashBuffer = hash.update(Buffer.from(txToSign, 'hex')).digest();
+
+    return { hashBuffer, derivationPath };
+  }
+
+  /**
+   * Gets the BitGo and user GPG keys from the BitGo public GPG key and the encrypted user GPG private key.
+   * @param {string} bitgoPublicGpgKey  - the BitGo public GPG key
+   * @param {string} encryptedUserGpgPrvKey  - the encrypted user GPG private key
+   * @param {string} walletPassphrase  - the wallet passphrase
+   * @returns {Promise<{ bitgoGpgKey: pgp.Key; userGpgKey: pgp.SerializedKeyPair<string> }>} - the BitGo and user GPG keys
+   */
+  private async getBitgoAndUserGpgKeys(
+    bitgoPublicGpgKey: string,
+    encryptedUserGpgPrvKey: string,
+    walletPassphrase: string
+  ): Promise<{
+    bitgoGpgKey: pgp.Key;
+    userGpgKey: pgp.SerializedKeyPair<string>;
+  }> {
+    const bitgoGpgKey = await pgp.readKey({ armoredKey: bitgoPublicGpgKey });
+    const userDecryptedKey = await pgp.readKey({
+      armoredKey: this.bitgo.decrypt({ input: encryptedUserGpgPrvKey, password: walletPassphrase }),
+    });
+    const userGpgKey: pgp.SerializedKeyPair<string> = {
+      privateKey: userDecryptedKey.armor(),
+      publicKey: userDecryptedKey.toPublic().armor(),
+    };
+    return {
+      bitgoGpgKey,
+      userGpgKey,
+    };
+  }
+
+  // #endregion
+
+  // #region external signer
+  /** @inheritdoc */
+  async signEcdsaMPCv2TssUsingExternalSigner(
+    params: TSSParams | TSSParamsForMessage,
+    externalSignerMPCv2Round1Generator: CustomMPCv2Round1GeneratingFunction,
+    externalSignerMPCv2Round2Generator: CustomMPCv2Round2GeneratingFunction,
+    externalSignerMPCv2Round3Generator: CustomMPCv2Round3GeneratingFunction,
+    requestType: RequestType = RequestType.tx
+  ): Promise<TxRequest> {
+    const { txRequest, reqId } = params;
+    let txRequestResolved: TxRequest;
+
+    assert(
+      requestType === RequestType.tx,
+      'Only transaction signing is supported for external signer, got: ' + requestType
+    );
```

Comment:
Lets add a ticket to support message signing with external signer. Don't want to let it fall through the cracks even though it is not needed atm.